libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -95,6 +106,8 @@ private void SetUpTurnState(ITurnContext turnContext, BotFrameworkClient botFram
             turnContext.TurnState.Add(_conversationState);
             turnContext.TurnState.Add(_userState);
             turnContext.TurnState.Add(_resourceExplorer);
+            turnContext.TurnState.Add(_memoryScopes);
+            turnContext.TurnState.Add(_pathResolvers);
             turnContext.TurnState.Add(_resourceExplorer.TryGetResource(_languageGeneratorId, out var resource) ? (LanguageGenerator)new ResourceMultiLanguageGenerator(_languageGeneratorId) : new TemplateEngineLanguageGenerator());
             turnContext.TurnState.Add(_languageGeneratorManagers.GetOrAdd(_resourceExplorer, _ => new LanguageGeneratorManager(_resourceExplorer)));
```

Comment:
can we do better with this existing LanaguageGenerator stuff?
handle it with the generic component mechanism your introducing here?

Reply:
John and I talked about this. We need them in DialogStateManager and the only connection to that world is turn state. Happy to hear other ideas!

Reply:
For a future PR :)